### DISCUSSION TITLE
will drift support both wasm for web and macros?

### DISCUSSION THREAD

**Discussion by Fraa-124** [simolus3/drift#2869] [General]
We want to incorporate (when the time comes) both macros and wasm into our web app, and we were wondering if drift will support both technologies. Using macros, would it be possible to use drift dart classes without using build_runner? For large projects, it takes a lot of time.

Thank you.

**Comment by simolus3** [comment]
Macros, dart2wasm and support for native assets with `dart:ffi` are all things that drift is aiming to fully support. At the moment, these are still in development and it's too soon for drift to start supporting them. However, I'm following the development work happening on the Dart SDK and I'm also making some contributions myself to ensure they will support everything drift will likely need to rely on them.

I don't have a timeline for these features, but you can expect drift to make use of them eventually. Especially the switch from `build_runner` to macros is a pretty big one though, so it will likely take some months to get every drift feature ready. But I'm experimenting with macros and drift already and we'll likely be able to offer a preview API once macros are becoming more stable.

**Reply by Fraa-124** [reply]
Glad to hear that, thanks.
